libpromises/cf3parse.y

```diff
@@ -755,10 +755,23 @@ litem:                 IDENTIFIER
                                        (P.currentclasses ?
                                             P.currentclasses : "any"),
                                        P.currentstring);
-                           RlistAppendScalar((Rlist **) &P.currentRlist,
-                                             (void *) P.currentstring);
-                           free(P.currentstring);
-                           P.currentstring = NULL;
+
+                           assert(!NULL_OR_EMPTY(P.lval));
+
+                           if (!StringEqual(P.lval + 1, "list") || P.lval[0] == 's')  // slists and everything else
```

Comment:
Please swap the two conditions, they will be easier to read and understand and also checking one character is faster/cheaper than checking a substring.